### PR TITLE
Reader: Search

### PR DESCRIPTION
**Pull request by kean**
- Fix an issue with search history overlapping the search results
- Fix an issue with "Posts" search straight up not working
- Fix an issue with the view occasionally requiring two taps instead of one to open the selected site/post (`backgroundTapRecognizer` was a fault)
- Use `searchController` to show a search bar with native look and feel
- Remove `Reader.storyboard`
- Reimplement search history using SwiftUI
- Remove unused code
- Remove some Jetpack migration code
- Add `Strings`

Screenshot 2024-11-05 at 1 42 23 PM Screenshot 2024-11-05 at 1 42 25 PM Screenshot 2024-11-05 at 1 42 27 PM

## Regression Notes
1. Potential unintended areas of impact

2. What I did to test those areas of impact (or what existing automated tests I relied on)

3. What automated tests I added (or what prevented me from doing so)

PR submission checklist:

- [ ] I have completed the Regression Notes.
- [ ] I have considered adding unit tests for my changes.
- [ ] I have considered adding accessibility improvements for my changes.
- [ ] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

Testing checklist:
- [ ] WordPress.com sites and self-hosted Jetpack sites.
- [ ] Portrait and landscape orientations.
- [ ] Light and dark modes.
- [ ] Fonts: Larger, smaller and bold text.
- [ ] High contrast.
- [ ] VoiceOver.
- [ ] Languages with large words or with letters/accents not frequently used in English.
- [ ] Right-to-left languages. (Even if translation isn’t complete, formatting should still respect the right-to-left layout)
- [ ] iPhone and iPad. 
- [ ] Multi-tasking: Split view and Slide over. (iPad)

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          2 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is larger than 500 lines of changes. Please consider splitting it into smaller PRs for easier and faster reviews.</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is assigned to the milestone <a href="https://github.com/wordpress-mobile/WordPress-iOS/milestone/297">25.5</a>. The due date for this milestone has already passed.<br />
Please assign it to a milestone with a later deadline or check whether the release for this milestone has already been finished.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: WordPress/Classes/Services/ReaderTopicService.m [hunk 1]
```diff
@@ -243,10 +243,6 @@ - (void)createSearchTopicForSearchPhrase:(NSString *)phrase completion:(void (^)
         [context obtainPermanentIDsForObjects:@[topic] error:nil];
         topicObjectID = topic.objectID;
     } completion:^{
-        // Save / update the search phrase to use it as a suggestion later.
-        ReaderSearchSuggestionService *suggestionService = [[ReaderSearchSuggestionService alloc] initWithCoreDataStack:self.coreDataStack];
```
Comment (kean) [inline WordPress/Classes/Services/ReaderTopicService.m:247]: It wasn't being invoked for site search, so I moved it to the search screen itself.

File: WordPress/Classes/ViewRelated/Reader/Controllers/ReaderSearchSuggestionsView.swift [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+                .search(in: allSuggestions, input: \.searchPhrase)
+        }
+    }
+
+    func delete(at indexSet: IndexSet) {
+        delete(indexSet.map { suggestions[$0] })
+    }
+
+    func delete(_ suggestions: [ReaderSearchSuggestion]) {
+        let context = coreData.mainContext
```
Comment (crazytonyli) [inline WordPress/Classes/ViewRelated/Reader/Controllers/ReaderSearchSuggestionsView.swift]: I'm aware that the original view controller implementation operates in the main context. But what do you think about using `ContextManager.performAndSave` to delete in the new implementation here?

Comment (crazytonyli) [reply WordPress/Classes/ViewRelated/Reader/Controllers/ReaderSearchSuggestionsView.swift]: Or, maybe not using Core Data at all.

Comment (kean) [reply WordPress/Classes/ViewRelated/Reader/Controllers/ReaderSearchSuggestionsView.swift]: > Or, maybe not using Core Data at all.

That was my original intention, but I decided to reduce the number of changes. I made some preparations by removing `NSFetchedResultsController`, so I was actually almost there. I ended up [removing](https://github.com/wordpress-mobile/WordPress-iOS/pull/23767/commits/34b82b4e4c44654b9d2e5e1fbe371925a1ac3c41) Core Data usage as you suggested.

Heads up: I forgot to merge and rebased on instead, so the history is updated.

**Review by crazytonyli — APPROVED** [review]

**Pull request merged**